testsuite/MDAnalysisTests/core/test_atomgroup.py

```diff
@@ -393,3 +512,779 @@ def test_split_VE(self):
+        ag = self.u.atoms[1000:1200]
+        ag.wrap(compound='segments')
+
+        cen = np.vstack([s.atoms.center_of_mass() for s in ag.segments])
+
+        assert_equal(self._in_box(cen), True)
+
+    def test_wrap_fragments(self):
+        ag = self.u.atoms[:250]
+        ag.wrap(compound='fragments')
+
+        cen = np.vstack([f.center_of_mass() for f in ag.fragments])
+
+        assert_equal(self._in_box(cen), True)
+
+
+class TestAtomGroupProperties(object):
+    """Test working with the properties of Atoms via AtomGroups
+
+    Check that:
+    - getting properties from AG matches the Atom values
+    - setting properties from AG changes the Atom
+    - setting the property on Atom changes AG
+    """
+    @staticmethod
+    def get_new(att_type):
+        """Return enough values to change the small g"""
+        if att_type == 'string':
+            return ['A', 'B', 'C', 'D', 'E', 'F']
+        elif att_type == 'float':
+            return np.array([0.001, 0.002, 0.003, 0.005, 0.012, 0.025], dtype=np.float32)
+        elif att_type == 'int':
+            return [4, 6, 8, 1, 5, 4]
+
+    def _check_ag_matches_atom(self, att, atts, ag):
+        """Checking Atomgroup property matches Atoms"""
+        # Check that accessing via AtomGroup is identical to doing
+        # a list comprehension over AG
+        ref = [getattr(atom, att) for atom in ag]
+
+        assert_equal(ref, getattr(ag, atts),
+                     err_msg="AtomGroup doesn't match Atoms for property: {0}".format(att))
+
+    def _change_atom_check_ag(self, att, atts, vals, ag):
+        """Changing Atom, checking AtomGroup matches this"""
+        # Set attributes via Atoms
+        for atom, val in zip(ag, vals):
+            setattr(atom, att, val)
+        # Check that AtomGroup returns new values
+        other = getattr(ag, atts)
+
+        assert_equal(vals, other,
+                     err_msg="Change to Atoms not reflected in AtomGroup for property: {0}".format(att))
+
+    @dec.skipif(parser_not_found('DCD'),
+                'DCD parser not available. Are you using python 3?')
+    def test_attributes(self):
+        u = make_Universe(('names', 'resids', 'segids', 'types', 'altLocs',
+                           'charges', 'masses', 'radii', 'bfactors',
+                           'occupancies'))
+        u.atoms.occupancies = 1.0
+        master = u.atoms
+        idx = [0, 1, 4, 7, 11, 14]
+        ag = master[idx]
+
+        for att, atts, att_type in (
+                ('name', 'names', 'string'),
+                ('type', 'types', 'string'),
+                ('altLoc', 'altLocs', 'string'),
+                ('charge', 'charges', 'float'),
+                ('mass', 'masses', 'float'),
+                ('radius', 'radii', 'float'),
+                ('bfactor', 'bfactors', 'float'),
+                ('occupancy', 'occupancies', 'float')
+        ):
+            vals = self.get_new(att_type)
+            yield self._check_ag_matches_atom, att, atts, ag
```

Comment:
I dislike this already. I find the yield test generators hard to follow in nose in general.